core/src/main/java/apoc/util/FileUtils.java

```diff
@@ -114,7 +115,10 @@ public boolean isEnabled() {
         }
 
         public URLStreamHandler createURLStreamHandler() {
-            return urlStreamHandlerFactory == null ? null : urlStreamHandlerFactory.createURLStreamHandler(this.name());
+            return Optional.ofNullable(urlStreamHandlerClassName)
+                    .map(Util::createInstanceOrNull)
+                    .map(i -> ((URLStreamHandlerFactory) i).createURLStreamHandler(this.name()))
```

Comment:
```suggestion
                    .map(urlStreamHandlerFactory -> urlStreamHandlerFactory.createURLStreamHandler(this.name()))
```

Non credo che il cast sia necessario

i

Reply:
Cambiato il nome della variabile. Il cast per come sta ora è necessario perché `.map(Util::createInstanceOrNull)` restituisce un `Optional<Object>`